src/openms/source/QC/FragmentMassError.cpp

```diff
@@ -72,7 +70,7 @@ namespace OpenMS
   }
 
 
-  PeakSpectrum getTheoSpec_(const Precursor::ActivationMethod& fm, const AASequence& seq, const int charge)
+  MSSpectrum FragmentMassError::getTheoSpec(const Precursor::ActivationMethod& fm, const AASequence& seq, const int charge)
```

Comment:
Of general interest so maybe move  that (static) function to TheoreticalSpectrumGenerator

Reply:
This function is a very simplified version of calculating a theoretical spectrum and probably only suited for a specific use case. (e.g. simplified ion type rule and simplified ion charge handling)
It might be a little confusing if it is a member of `TheoreticalSpectrumGenerator`.

Reply:
Maybe it would be better to move it to `QCBase`?

Reply:
Hmm I was thinking in lines of e.g. a pyOpenMS user that just wants to generate the typical ions for HCD fragmentation and a peptide. For this your function would be ideal and putting it in the TheoreticalSpectrumGenerator namespace would make it easily findable. I actually thought about adding such a function before and I was super happy that you already did it :P
Changing and refining it later to e.g. produce let's say other less common ions (like immonium ions) could be added without interfering with your code I guess? You would still be able to calculate fragment mass errors.

Reply:
How about the fragment charge? The rule here is also quite simple.

Reply:
1 to precursor charge - 1 is totally fine in my view for most use cases so that is what I would expect form such a method

Reply:
and probably rename it if moved to TheoreticalSpectrumGenerator.
TheoreticalSpectrumGenerator::generateSpectrum( ...)
would be a good name

Reply:
alright :)